src/IniWriter.php

```diff
@@ -93,8 +93,12 @@ public function writeToString(array $config, $header = '')
                 }
 
                 if (is_array($value)) {
-                    foreach ($value as $currentValue) {
-                        $ini .= $option . '[] = ' . $this->encodeValue($currentValue) . "\n";
+                    foreach ($value as $key => $currentValue) {
+                        if (is_int($key)) {
+                            $ini .= $option . '[] = ' . $this->encodeValue($currentValue) . "\n";
+                        } else {
+                            $ini .= $option . '[' . $key . '] = ' . $this->encodeValue($currentValue) . "\n";
```

Comment:
I haven't properly reviewed but wondering if `$key` would need to be encoded as well maybe? Just thinking randomly if `$key` was to be supplied by user input then not sure if you could break something or cause various security issues eg by defining a key like `"] = 'xyz';\n foo ="`.

Reply:
I'll check it up and write a test :)

Reply:
`encodeValue()` is not needed for the `$key`. It's just for `booleans` and **quoted** `strings` as value.

Reply:
I think what @tsteur means is, can you write a test where the key is defined as `"] = 'xyz';\n foo ="` and make sure it doesn't inject something into the INI file?

Reply:
@diosmosis, I'll write this test today.

Reply:
The `$key` will be sanitized now. But you can still break the INI by injecting malicious crap `"] = 'xyz';\n foo ="` to the INI section or as value :/

Reply:
Thanks for writing the test @typomedia! I left one other comment, after that it will be good to merge. If you'd also like to make sure this doesn't happen for sections or values, we'd certainly appreciate it! Otherwise we'll take a look ourselves after merging this PR.